src/Blocks/Expandable.purs

```diff
@@ -2,16 +2,39 @@ module Ocelot.Block.Expandable where
 
 import Prelude
 
-import DOM.HTML.Indexed (HTMLheader, HTMLspan, HTMLdiv)
+import DOM.HTML.Indexed (HTMLdiv, HTMLheader, HTMLspan, Interactive)
+import Data.Array (snoc)
+import Data.Bifunctor (lmap, rmap)
+import Data.Foldable (foldr)
+import Data.Maybe (Maybe(..), fromMaybe)
+import Data.Tuple (Tuple(..))
+import Halogen.HTML (PropName(..))
 import Halogen.HTML as HH
+import Halogen.HTML.Core (class IsProp, Prop(..), PropValue)
 import Halogen.HTML.Properties as HP
+import Halogen.VDom.DOM.Prop (propFromString)
 import Ocelot.Block.Icon as Icon
 import Ocelot.HTML.Properties ((<&>))
+import Unsafe.Coerce (unsafeCoerce)
 
 data Status
   = Collapsed
   | Expanded
 
+renderStatus :: Status -> String
+renderStatus = case _ of
+  Collapsed -> "collapsed"
+  Expanded  -> "expanded"
+
+readStatus :: String -> Maybe Status
```

Comment:
We rely on `purescript-read` which exposes a `Read` type class that should be used here.